browser/src/slideshow/LayerDrawing.ts

```diff
@@ -504,27 +516,43 @@ class LayerDrawing {
 			return;
 		}
 
-		if (info.index === 0 || !this.cachedMasterPages.get(slideInfo.masterPage))
-			this.cachedMasterPages.set(slideInfo.masterPage, new Array<LayerEntry>());
-
-		const layers = this.cachedMasterPages.get(slideInfo.masterPage);
-		if (layers.length !== info.index) {
-			window.app.console.log(
-				'LayerDrawing.handleMasterPageLayerMsg: missed any layers ?',
-			);
-		}
 		const layerEntry: LayerEntry = {
 			type: info.type,
 			content: info.content,
 			isField: info.isField,
 		};
-		if (info.type === 'bitmap') {
-			if (!this.checkAndAttachImageData(layerEntry.content as ImageInfo, img))
-				return;
-		}
-		layers.push(layerEntry);
 
-		this.drawMasterPageLayer(layerEntry, info.slideHash);
+		var drawImage = () => {
+			if (info.index === 0 || !this.cachedMasterPages.get(slideInfo.masterPage))
+				this.cachedMasterPages.set(
+					slideInfo.masterPage,
+					new Array<LayerEntry>(),
+				);
+
+			const layers = this.cachedMasterPages.get(slideInfo.masterPage);
+			if (layers.length !== info.index) {
+				window.app.console.log(
+					'LayerDrawing.handleMasterPageLayerMsg: missed any layers ?',
+				);
+			}
+
+			layers.push(layerEntry);
+
+			this.drawMasterPageLayer(layerEntry, info.slideHash);
+		};
+		if (img.rawData.length !== 0 && info.type === 'bitmap') {
+			var img_ = (window as any).fzstd.decompress(img.rawData);
+			const clampedthing = new Uint8ClampedArray(img_);
+			var imageData = new ImageData(clampedthing, info.width, info.height);
```

Comment:
I wonder if we can at least make these duplicate three lines img_=, clampedthing=, imageData= split out into a function that returns ImageData from an arg of rawData and use that everywhere instead (and rename clampedthing to something without "thing" in it)